File: problems/1337_the_k_weakest_rows_in_a_matrix/solution.cpp
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <algorithm>
#include <climits>
#include <queue>
#include <stack>
#include <cmath>
#include <numeric>
#include <set>
#include <map>
using namespace std;

class Solution {
public:
    vector<int> kWeakestRows(vector<vector<int>>& mat, int k) {
        vector<pair<int,int>> strengths;
        for (int i = 0; i < (int)mat.size(); i++)
            strengths.push_back({accumulate(mat[i].begin(), mat[i].end(), 0), i});
        sort(strengths.begin(), strengths.end());
        vector<int> res;
        for (int i = 0; i < k; i++) res.push_back(strengths[i].second);
        return res;
    }
};

File: problems/1337_the_k_weakest_rows_in_a_matrix/Solution.java
import java.util.*;

class Solution {
    public int[] kWeakestRows(int[][] mat, int k) {
        int m = mat.length;
        int[][] strengths = new int[m][2];
        for (int i = 0; i < m; i++) {
            strengths[i][0] = 0;
            for (int x : mat[i]) strengths[i][0] += x;
            strengths[i][1] = i;
        }
        Arrays.sort(strengths, (a, b) -> a[0] != b[0] ? a[0] - b[0] : a[1] - b[1]);
        int[] res = new int[k];
        for (int i = 0; i < k; i++) res[i] = strengths[i][1];
        return res;
    }
}
